src/taipy/core/_repository/_sql_repository.py

```diff
@@ -139,17 +139,15 @@ def _get_by_configs_and_owner_ids(self, configs_and_owner_ids):
         configs_and_owner_ids = set(configs_and_owner_ids)
 
         for config, owner in configs_and_owner_ids:
-            entry = self.__get_entities_by_config_and_owner(config.id, owner, only_first=True)
+            entry = self.__get_entities_by_config_and_owner(config.id, owner)
             if entry:
                 entity = self.__to_entity(entry)
                 key = config, owner
                 res[key] = entity
 
         return res
 
-    def __get_entities_by_config_and_owner(
-        self, config_id: str, owner_id: Optional[str] = "", only_first: bool = False
-    ):
+    def __get_entities_by_config_and_owner(self, config_id: str, owner_id: Optional[str] = "") -> _TaipyModel:
```

Comment:
*Note to reviewers* This function is called only to retrieve one single entry element. As it is private, it is safe to just always call `query.first()` and then drop the `only_first` flag.